go/worker/compute/executor/committee/node.go

```diff
@@ -1314,24 +1314,23 @@ func (n *Node) HandleNewEventLocked(ev *roothash.Event) {
 
 // Guarded by n.commonNode.CrossNode.
 func (n *Node) handleExternalBatchLocked(batch *unresolvedBatch) error {
-	// If we are not waiting for a batch, don't do anything.
+	// If we are not waiting for a batch, don't do anything but still forward proposal.
 	if _, ok := n.state.(StateWaitingForBatch); !ok {
-		return errIncorrectState
+		return nil // Forward proposal.
 	}
 
 	epoch := n.commonNode.Group.GetEpochSnapshot()
 
-	// We can only receive external batches if we are an executor member.
+	// Ignore proposals if we are not an executor committee member, but still forward them.
 	if !epoch.IsExecutorMember() {
-		n.logger.Error("got external batch while in incorrect role")
-		return errIncorrectRole
+		return nil // Forward proposal.
```

Comment:
Before, in these cases where a non-permanent p2p error was returned the dispatch was retried right? 

Is it ok that there is now no retry? (e.g. if a node is lagging behind, it could be able to process this message in a followup retry?).

Reply:
No, I already changed this back in my local branch but it turns out that actually this retry process is currently implemented incorrectly as the retry could time out just before the proposal would become valid. I'm fixing it locally and will push a new version soon.

Reply:
Just pushed the new version, handling of proposal retries is now done in the executor as it knows when it makes sense to retry (e.g. after processing a new block update).

Reply:
:+1: the new version makes sense to me